src-ts/tools/learn/learn-lib/my-certifications-provider/my-certification-progress.provider.tsx

```diff
@@ -19,17 +22,19 @@ export const useMyCertificationProgress: (certification?: string) => MyCertifica
             loading: true,
         }))
 
-        const t: ReturnType<typeof setTimeout> = setTimeout(() => {
+        if (!userId) {
+            return
+        }
+
+        getMyCertificationsProgressAsync(userId, certification).then(mapCompletedPercentage).then((myCertifications) => {
             setState((prevState) => ({
                 ...prevState,
+                certificateProgress: myCertifications.find(c => c.certification === certification),
```

Comment:
I think this should be using the `certificationProgressId` instead of the certification name so that we are 100% sure we have the right one.

Reply:
Hmm. This is a similar issue as on the CourseDetailsPage. We don't have access to the certificationProgressId in here, but when fetching the `myCertifications`, I'm now sending the provider as well. Basically myCertifications is a very specific response and it should always return 1 certificateProgress. I'm doing the "find" filtering just to be sure and to not use `myCertifications[0]`.